packages/util/src/verkle.ts

```diff
@@ -0,0 +1,228 @@
+  }
+
+  const treeStem = ffi.getTreeKey(address32, treeIndexBytes, 0).slice(0, 31)
+
+  return treeStem
+}
+
+/**
+ * Verifies that the executionWitness is valid for the given prestateRoot.
+ * @param ffi The verkle ffi object from verkle-crypotography-wasm.
+ * @param prestateRoot The prestateRoot matching the executionWitness.
+ * @param executionWitness The verkle execution witness.
+ * @returns {boolean} Whether or not the executionWitness belongs to the prestateRoot.
+ */
+export function verifyProof(
+  ffi: VerkleCrypto,
+  prestateRoot: Uint8Array,
+  executionWitness: VerkleExecutionWitness
+): boolean {
+  return ffi.verifyExecutionWitnessPreState(
+    bytesToHex(prestateRoot),
+    JSON.stringify(executionWitness)
+  )
+}
+
+/* Verkle Structure */
+
+export interface VerkleProof {
+  commitmentsByPath: PrefixedHexString[]
+  d: PrefixedHexString
+  depthExtensionPresent: PrefixedHexString
+  ipaProof: {
+    cl: PrefixedHexString[]
+    cr: PrefixedHexString[]
+    finalEvaluation: PrefixedHexString
+  }
+  otherStems: PrefixedHexString[]
+}
+
+export interface VerkleStateDiff {
+  stem: PrefixedHexString
+  suffixDiffs: {
+    currentValue: PrefixedHexString | null
+    newValue: PrefixedHexString | null
+    suffix: number | string
+  }[]
+}
+
+/**
+ * Experimental, object format could eventual change.
+ * An object that provides the state and proof necessary for verkle stateless execution
+ * */
+export interface VerkleExecutionWitness {
+  /**
+   * An array of state diffs.
+   * Each item corresponding to state accesses or state modifications of the block.
+   * In the current design, it also contains the resulting state of the block execution (post-state).
+   */
+  stateDiff: VerkleStateDiff[]
+  /**
+   * The verkle proof for the block.
+   * Proves that the provided stateDiff belongs to the canonical verkle tree.
+   */
+  verkleProof: VerkleProof
+}
+
+export enum LeafType {
+  Version = 0,
+  Balance = 1,
+  Nonce = 2,
+  CodeHash = 3,
+  CodeSize = 4,
+}
+
+export const VERSION_LEAF_KEY = intToBytes(LeafType.Version)
+export const BALANCE_LEAF_KEY = intToBytes(LeafType.Balance)
+export const NONCE_LEAF_KEY = intToBytes(LeafType.Nonce)
+export const CODE_HASH_LEAF_KEY = intToBytes(LeafType.CodeHash)
+export const CODE_SIZE_LEAF_KEY = intToBytes(LeafType.CodeSize)
+
+export const HEADER_STORAGE_OFFSET = 64
+export const CODE_OFFSET = 128
+export const VERKLE_NODE_WIDTH = 256
+export const MAIN_STORAGE_OFFSET = BigInt(256) ** BigInt(31)
+
+/**
+ * @dev Returns the tree key for a given verkle tree stem, and sub index.
+ * @dev Assumes that the verkle node width = 256
+ * @param stem The 31-bytes verkle tree stem as a Uint8Array.
+ * @param subIndex The sub index of the tree to generate the key for as a Uint8Array.
+ * @return The tree key as a Uint8Array.
+ */
+
+export const getKey = (stem: Uint8Array, leaf: LeafType | Uint8Array) => {
```

Comment:
Minor note on a few of these helpers. They were initially named quite minimally `getKey` rather than `getVerkleTreekey` because the "verkle" was obvious from it being imported from the verkle package. I feel like we should likely rename these helpers more explicitly as "verkle" since they are now exported in utils (`getKey` could be so many things). We could perhaps do the same with `VERSION_LEAF_KEY` => `VERKLE_VERSION_LEAF_KEY`